secboot/encrypt_sb.go

```diff
@@ -49,7 +49,14 @@ const metadataKiBSize = 2048     // 2MB
 // FormatEncryptedDevice initializes an encrypted volume on the block device
 // given by node, setting the specified label. The key used to unlock the volume
 // is provided using the key argument.
-func FormatEncryptedDevice(key keys.EncryptionKey, label, node string) error {
+func FormatEncryptedDevice(key keys.EncryptionKey, encType EncryptionType, label, node string) error {
+	if !encType.IsLUKS() {
+		return fmt.Errorf("cannot use encryption type %q when formating device %q", encType, node)
```

Comment:
maybe: "internal error: FormatEncryptedDevice for %q expects a LUKS encryption type, not %q", node, encType